File: Backend/MeetSpace/MeetSpace.Api/Contracts/Bookings/GetBookingResponse.cs
using MeetSpace.Api.Contracts.Rooms;
using MeetSpace.Api.Contracts.Users;
using MeetSpace.Domain.Models;

namespace MeetSpace.Api.Contracts.Bookings;

public class GetBookingResponse
{
    public Guid Id { get; set; }
    public GetRoomResponse Room { get; set; } = null!;
    public Guid UserId { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public TimeOnly StartOfBookingTime { get; set; }
    public TimeOnly EndOfBookingTime { get; set; }
    public DateOnly BookingDate { get; set; }
}

File: Frontend/src/models/GetBookingResponse.ts
import { BookingRoom } from '@/models/BookingRoom';

export interface GetBookingResponse {
    id: string;
    room: BookingRoom;
    userId: string;
    title: string;
    description?: string;
    bookingDate: string;
    startOfBookingTime: string;
    endOfBookingTime: string;
}
